File: app/src/main/java/org/xhanka/biblesiswati/settings/SettingsFragment.java
package org.xhanka.biblesiswati.settings;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.SwitchPreferenceCompat;

import org.jetbrains.annotations.NotNull;
import org.xhanka.biblesiswati.R;
import org.xhanka.biblesiswati.common.Utils;

public class SettingsFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreate(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.root_preferences, rootKey);

        SwitchPreferenceCompat preference = findPreference("dark_mode");

        if (preference != null) {
            preference.setChecked(Utils.Companion.getDarkMode(getContext()));

            preference.setOnPreferenceChangeListener((preference1, newValue) -> {
                Utils.Companion.handleDarkMode((boolean) newValue, getContext());
                return true;
            });
        }
    }

    @Override
    public void onCreateOptionsMenu(@NonNull @NotNull Menu menu, @NonNull @NotNull MenuInflater inflater) {
        menu.clear();
        super.onCreateOptionsMenu(menu, inflater);
    }
}

File: app/src/main/java/org/xhanka/biblesiswati/settings/SettingsFragment.kt
package org.xhanka.biblesiswati.settings

import android.os.Bundle
import android.view.Menu
import android.view.MenuInflater
import androidx.preference.ListPreference
import androidx.preference.PreferenceFragmentCompat
import org.xhanka.biblesiswati.R
import org.xhanka.biblesiswati.common.Constants
import org.xhanka.biblesiswati.common.Utils

class SettingsFragment : PreferenceFragmentCompat() {
    override fun onCreate(savedInstanceState: Bundle?) {
        setHasOptionsMenu(true)
        super.onCreate(savedInstanceState)
    }

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.root_preferences, rootKey)

        val darkMode: ListPreference? = findPreference(Constants.DARK_MODE)
        darkMode?.let { it ->
            it.setOnPreferenceChangeListener { _, newValue ->
                Utils.setDarkMode(newValue as String)
                true
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        menu.clear()
        super.onCreateOptionsMenu(menu, inflater)
    }
}